include/libdnf/rpm/repo.hpp

```diff
@@ -1,112 +1,103 @@
+    const char * get_domain_name() const noexcept override { return "librepo"; }
+    const char * get_name() const noexcept override { return "LrException"; }
+    const char * get_description() const noexcept override { return "Librepo exception"; }
+    int get_code() const noexcept { return code; }
+
 private:
     int code;
 };
 
-/**
-* @class RepoCB
-*
-* @brief Base class for Repo callbacks
-*
-* User implements repo callbacks by inheriting this class and overriding its methods.
-*/
+/// Base class for Repo callbacks
+/// User implements repo callbacks by inheriting this class and overriding its methods.
 class RepoCB {
 public:
     enum class FastestMirrorStage {
-        INIT, /*!<
-            Fastest mirror detection just started.
-            ptr is NULL*/
-
-        CACHELOADING, /*!<
-            ptr is (char *) pointer to string with path to the cache file.
-            (Do not modify or free the string). */
-
-        CACHELOADINGSTATUS, /*!<
-            if cache was loaded successfully, ptr is NULL, otherwise
-            ptr is (char *) string with error message.
-            (Do not modify or free the string) */
-
-        DETECTION, /*!<
-            Detection (pinging) in progress.
-            If all data was loaded from cache, this stage is skiped.
-            ptr is pointer to long. This is the number of how much
-            mirrors have to be "pinged" */
-
-        FINISHING, /*!<
-            Detection is done, sorting mirrors, updating cache, etc.
-            ptr is NULL */
-
-        STATUS, /*!<
-            The very last invocation of fastest mirror callback.
-            If fastest mirror detection was successful ptr is NULL,
-            otherwise ptr contain (char *) string with error message.
-            (Do not modify or free the string) */
+        /// Fastest mirror detection just started. ptr is NULL
+        INIT,
+
+        /// ptr is (char *) pointer to string with path to the cache file. Do not modify or free the string.
+        CACHELOADING,
+
+        /// if cache was loaded successfully, ptr is NULL, otherwise ptr is (char *) string with error message.
+        /// Do not modify or free the string.
+        CACHELOADINGSTATUS,
+
+        /// Detection (pinging) in progress.
+        /// If all data was loaded from cache, this stage is skiped. ptr is pointer to long.
+        /// This is the number of how much mirrors have to be "pinged"
+        DETECTION,
+
+        /// Detection is done, sorting mirrors, updating cache, etc. ptr is NULL
+        FINISHING,
+
+        /// The very last invocation of fastest mirror callback.
+        /// If fastest mirror detection was successful ptr is NULL, otherwise ptr contain (char *) string with error message.
+        /// Do not modify or free the string.
+        STATUS,
     };
 
-    virtual void start(const char *what) {}
+    virtual void start(const char * /*what*/) {}
```

Comment:
I am sorry, is it correct?

Reply:
Probably note could be added.

Reply:
replace with
```virtual void start([[maybe_unused]] const char * what)```
It is supported in c++17 and seems to work fine with both gcc and clang

Reply:
I tried C++ attributes some time ago. And there were problem with swig.

Reply:
I have an idea. Almost all occurrences are in the implementation files. Swig parses only public headers. And there is only one occurrence in repo.hpp. I will move the implementation to .cpp . And I will use `[[maybe_unused]]` attribute there.